test/utils/definePropertyAtPath.test.ts

```diff
@@ -15,11 +15,31 @@ describe('definePropertyAtPath()', () => {
 
   it('defines a nested property at a given path', () => {
     const target: AnyObject = {}
-    definePropertyAtPath(target, 'a.b.c', {
+    definePropertyAtPath(target, ['a', 'b', 'c'], {
```

Comment:
This is perhaps a silly question, but `propertyPath` here is not necessarily an array of strings, right?

If our intention is to use this paths for `lodash.get`/`lodash.set`, then it can also reference array indices in the path:

```js
definePropertyAtPath(target, ['a', 2, 'b'], value)
```

Numbers are also valid object keys, which makes me wonder if we should introduce a `PropertyPath` custom type that's a union of `string` and `number`:

```js
export type PropertyPath = Array<string | number>
```

What do you think about this, @VanTanev?

Reply:
No, by ECMA standards an object key is ALWAYS a string. It will be cast to a string, regardless of what it was initially.

eg:
```ts
const obj = {}
const test = { 1: 'a', [obj]: 'b' }

Object.keys(test)
// ['1', '[object Object]']

test[1]
// a

test['1']
// a

test[obj]
// 'b'

test[{}]
// 'b'

test['[object Object]']
// 'b'
```

With that in mind, enforcing string keys in the type is good in my opinion, given that any value will ultimately be cast to a string. It's more straightforward and makes these implicit casts explicit.

Reply:
Sorry, I just noticed I skipped a part of your question about array indices. 

It's true that `lodash.set`/`lodash.get` can set array indices, and numbers are valid there. However, my understanding is that `definePropertyAtPath` is meant to be used only for hashmap objects, and not for arrays, given the use of `Object.defineProperty()`.

Reply:
At least not as of now. Using strict string type for object keys makes sense, thank you for explaining that.